param/parameterized.py

```diff
@@ -553,7 +564,8 @@ class Foo(Bar):
 
     def __init__(self,default=None,doc=None,precedence=None,  # pylint: disable-msg=R0913
                  instantiate=False,constant=False,readonly=False,
-                 pickle_default_value=True, allow_None=False):
+                 pickle_default_value=True, allow_None=False,
+                 per_instance=True):
```

Comment:
The `per_instance` flag should be described in the docstring below. What I'm wondering is when a user should set this flag. I can imagine certain parameter *types* shouldn't allow `per_instance` but in that case, why expose this through the constructor? If a user disabled `per_instance` when setting a parameter, couldn't this have confusing repercussions if the watch mechanism is used for example?

Reply:
You might genuinely want to share parameter attributes across multiple instances of a class.